packages/taco/src/taco.ts

```diff
@@ -1,34 +1,88 @@
-import { DkgPublicKey, ThresholdMessageKit } from '@nucypher/nucypher-core';
 import {
-  Condition,
-  ConditionExpression,
-  DkgClient,
+  AccessControlPolicy,
+  DkgPublicKey,
+  encryptForDkg,
+  ThresholdMessageKit,
+} from '@nucypher/nucypher-core';
+import {
   DkgCoordinatorAgent,
-  Enrico,
+  fromHexString,
   getPorterUri,
-  ThresholdDecrypter,
   toBytes,
 } from '@nucypher/shared';
 import { ethers } from 'ethers';
+import { keccak256 } from 'ethers/lib/utils';
+
+import { Condition, ConditionExpression } from './conditions';
+import { DkgClient } from './dkg';
+import { retrieveAndDecrypt } from './tdec';
 
 export const encrypt = async (
   provider: ethers.providers.Provider,
-  message: string,
+  message: Uint8Array | string,
   condition: Condition,
   ritualId: number,
+  authSigner: ethers.Signer,
 ): Promise<ThresholdMessageKit> => {
+  // TODO(#264): Enable ritual initialization
+  // if (ritualId === undefined) {
+  //   ritualId = await DkgClient.initializeRitual(
+  //     provider,
+  //     this.cohort.ursulaAddresses,
+  //     true
+  //   );
+  // }
+  // if (ritualId === undefined) {
+  //   // Given that we just initialized the ritual, this should never happen
+  //   throw new Error('Ritual ID is undefined');
+  // }
   const dkgRitual = await DkgClient.getFinalizedRitual(provider, ritualId);
-  return await encryptLight(message, condition, dkgRitual.dkgPublicKey);
+
+  const mk = await encryptWithPublicKey(
+    message,
+    condition,
+    dkgRitual.dkgPublicKey,
+    authSigner,
+  );
+
+  const isAuthorized = DkgCoordinatorAgent.isEncryptionAuthorized(
```

Comment:
I think this check should be optional, and **not done by default**.

In some/most cases it is ok for Enrico to encrypt before being authorized. Authorization can be granted after encryption and the authorization check only really applies at decryption time (this was a prior question from BqEth).

I get the notion of being proactive about authorization before encryption, but they are different in their timing.

(cc @cygnusv , @KPrasch ).

Reply:
Indeed, it's use-case specific, ritual-specific, and may actually be performed by different actors or user stories altogether.